client/src/api/API.res

```diff
@@ -261,6 +261,39 @@ let saveTest = (m: model): cmd =>
     x,
   ))
 
+let fetchServerBuildHash = (m: model): cmd => {
+  let url = "https://editor.darklang.com/latest-backend-build-hash"
+
+  let request = Tea.Http.request({
+    method: "POST",
+    headers: list{clientVersionHeader(m), Header("X-CSRF-Token", m.csrfToken)},
+    url: url,
+    body: Web.XMLHttpRequest.EmptyBody,
+    expect: Tea.Http.expectStringResponse(Decoders.wrapExpect(Json.Decode.string)),
+    timeout: None,
+    withCredentials: false,
+  })
+
+  // If origin is https://darklang.com, then we're in prod (or ngrok, running against
+  // prod) and editor.darklang.com's CORS rules will allow this request. If not,
+  // we're in local, and both CORS and auth (session, canvas_id) will not work
+  // against editor.darklang.com. By putting the conditional here instead of at the
+  // beginning of the function, we still exercise the message and request generating
+  // code locally.
+  if m.origin == "https://darklang.com" {
+    // TODO: test this locally.
```

Comment:
My local testing so far has been to raise the `RefreshClientIfOutdated` Msg with a known-incorrect build hash. I suppose the next step, to better test, will be to set up a tunnel.